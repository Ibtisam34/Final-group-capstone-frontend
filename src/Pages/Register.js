import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import * as Yup from 'yup';
import { Formik, Form } from 'formik';
// import userToken from '../redux/reducer/user/userToken';
import { signUp } from '../redux/reducer/user/userSlice';

const Register = () => {
  const initialValues = {
    name: '',
    email: '',
    password: '',
    password_confirmation: '',
  };

  const SignUpSchema = Yup.object().shape({
    name: Yup.string()
      .min(1, 'Too Short!')
      .max(100, 'Too Long')
      .matches(
        /^(?=.{4,50}$)(?![a-z])(?!.*[_.]{2})[a-zA-Z ]+(?<![_.])$/,
        'Username should have at least 4 characters and should not contain numbers or special characters/punctuations!',
      ),
    email: Yup.string()
      .email('Invalid Email!'),
    password: Yup.string()
      .matches(
        /^[a-zA-Z0-9!@#$%^&* ]{6,20}$/,
        'Password must contain at least 6 characters!',
      ),
    password_confirmation: Yup.string()
      .oneOf([Yup.ref('password'), null], 'Password not match!'),
  });

  const dispatch = useDispatch();

  const navigate = useNavigate();

  const [values, setValues] = useState(initialValues);
  const [signupStatus, setSignupStatus] = useState('');

  const handleChange = (event) => {
    const { name, value } = event.target;
    setValues((prevState) => ({ ...prevState, [name]: value }));
  };

  const signUphandler = async (user) => {
    const register = await dispatch(signUp(user));
    if (register.payload.status.code === 200) {
      localStorage.setItem('authUser', JSON.stringify(register.payload.data));
      setSignupStatus('success');
      setValues((prevState) => ({
        ...prevState,
        name: '',
        email: '',
        password: '',
        password_confirmation: '',
      }));
      navigate('/');
    } else {
      setSignupStatus('error');
    }
  };

  document.title = 'Luxury Cars | Register';
  return (
    <>
      {/* <Alert /> */}
      <div className="container-fluid mt-5">
        <div className="row justify-content-center">
          <div className="col-md-6">
            <div className="card">
              <div className="card-header">
                Register
              </div>
              <div className="card-body">
                {signupStatus === 'success' && (
                  <div className="alert alert-success" role="alert">
                    User successfully registered!
                  </div>
                )}
                {signupStatus === 'error' && (
                  <div className="alert alert-danger" role="alert">
                    An error occured!
                  </div>
                )}
                <Formik
                  initialValues={initialValues}
                  validationSchema={SignUpSchema}
                  onSubmit={() => signUphandler({
                    name: values.name,
                    email: values.email,
                    password: values.password,
                  })}
                >
                  {({
                    errors, touched,
                  }) => (
                    <Form>
                      <div className="form-group mb-3">
                        <label htmlFor="nameInput" className="form-label" style={{ width: '100%' }}>
                          Name
                          <input
                            type="text"
                            name="name"
                            className="form-control"
                            id="nameInput"
                            autoComplete="name"
                            value={values.name}
                            onChange={handleChange}
                          />
                        </label>
                        {errors.name && touched.name && <div className="form-error">{errors.name}</div>}
                      </div>
                      <div className="form-group mb-3">
                        <label htmlFor="emailInput" className="form-label" style={{ width: '100%' }}>
                          Email address
                          <input
                            type="email"
                            name="email"
                            className="form-control"
                            id="emailInput"
                            autoComplete="email"
                            value={values.email}
                            onChange={handleChange}
                          />
                        </label>
                        {errors.email && touched.email && <div className="form-error">{errors.email}</div>}
                      </div>
                      <div className="form-group mb-3">
                        <label htmlFor="passwordInput" className="form-label" style={{ width: '100%' }}>
                          Password
                          <input
                            type="password"
                            name="password"
                            className="form-control"
                            id="passwordInput"
                            autoComplete="password"
                            value={values.password}
                            onChange={handleChange}
                          />
                        </label>
                        {errors.password && touched.password && <div className="form-error">{errors.password}</div>}
                      </div>
                      <div className="form-group mb-3">
                        <label htmlFor="confirmPasswordInput" className="form-label" style={{ width: '100%' }}>
                          Confirm password
                          <input
                            type="password"
                            name="password_confirmation"
                            className="form-control"
                            id="confirmPasswordInput"
                            autoComplete="password"
                            value={values.password_confirmation}
                            onChange={handleChange}
                          />
                        </label>
                        {errors.confirmPassword && touched.confirmPassword && (
                          <div className="form-error">{errors.confirmPassword}</div>
                        )}
                      </div>
                      <button type="submit" className="btn btn-primary">
                        Register
                      </button>
                    </Form>
                  )}
                </Formik>
                <div className="mt-3">
                  Already have an account?
                  {' '}
                  <a href="/login">Login</a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default Register;
