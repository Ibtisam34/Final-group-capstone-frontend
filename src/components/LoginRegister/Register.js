import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Slider from './Slider';
import Logo from '../../assets/images/LOGO-DOCTOR.png';
import { postUsers } from '../../redux/users/usersReducer';

function Register() {
  const dispatch = useDispatch();
  const [userName, setName] = useState('');
  const [userEmail, setEmail] = useState('');

  const navigate = useNavigate();

  const createUser = (e) => {
    e.preventDefault();
    if (userName !== '' && userEmail !== '') {
      const postData = {
        user:
        {
          name: userName,
          email: userEmail,
        },
      };
      dispatch(postUsers(postData));
    }
  };

  const users = useSelector((state) => state.usersReducer);
  return (
    <Container fluid>
      <Container className="login-card col-9 pb-5 bg-white border rounded">
        <div className="logo-wrapper">
          <img src={Logo} alt="logo" className="logo-img" width="200" height="200" />
        </div>
        <Form>
          <Form.Group className="mt-3 mb-3" controlId="formBasicName">
            <Form.Control type="text" placeholder="Full name" onChange={(e) => setName(e.target.value)} required />
          </Form.Group>

          <Form.Group className="mb-3" controlId="formBasicEmail">
            <Form.Control type="email" placeholder="Email address" onChange={(e) => setEmail(e.target.value)} required />
            <Form.Text className="text-muted">
              We will never share your email with anyone else.
            </Form.Text>
          </Form.Group>

          <Form.Group className="mb-3 text-center">
            <Button variant="success" type="submit" onClick={(e) => createUser(e)}>
              Register
            </Button>
          </Form.Group>
          { users.status === 201
            && <p className="text-success">Registration was successful!</p>}
        </Form>
        <div>
          <p>
            Already have an account?
            {' '}
            <span>
              <button
                type="button"
                onClick={() => { navigate('/login'); }}
                className="access-btn text-primary"
              >
                Login
              </button>
            </span>
          </p>
        </div>
      </Container>
      <Slider />
    </Container>
  );
}

export default Register;
