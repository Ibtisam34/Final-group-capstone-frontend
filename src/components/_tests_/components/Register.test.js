import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/configureStore';
import Register from '../../LoginRegister/Register';

describe('Register to app', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><Register /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
