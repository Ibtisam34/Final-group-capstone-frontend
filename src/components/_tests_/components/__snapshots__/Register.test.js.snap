// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register to app renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <div
          class="login-card col-4 pb-5 bg-white border rounded container"
        >
          <div
            class="logo-wrapper"
          >
            <img
              alt="logo"
              class="logo-img"
              height="200"
              src="LOGO-DOCTOR.png"
              width="200"
            />
          </div>
          <form
            class=""
          >
            <div
              class="mt-3 mb-3"
            >
              <input
                class="form-control"
                id="formBasicName"
                placeholder="Full name"
                required=""
                type="text"
              />
            </div>
            <div
              class="mb-3"
            >
              <input
                class="form-control"
                id="formBasicEmail"
                placeholder="Email address"
                required=""
                type="email"
              />
              <small
                class="text-muted form-text"
              >
                We will never share your email with anyone else.
              </small>
            </div>
            <div
              class="mb-3 text-center"
            >
              <button
                class="btn btn-success"
                type="submit"
              >
                Register
              </button>
            </div>
          </form>
          <div>
            <p>
              Already have an account?
               
              <span>
                <button
                  class="access-btn text-primary"
                  type="button"
                >
                  Login
                </button>
              </span>
            </p>
          </div>
        </div>
        <div
          class="carousel carousel slide"
        >
          <div
            class="carousel-indicators"
          >
            <button
              aria-current="true"
              aria-label="Slide 1"
              class="active"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 2"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 3"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 4"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 5"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 6"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 7"
              data-bs-target=""
              type="button"
            />
          </div>
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item active carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://images.unsplash.com/photo-1629909613654-28e377c37b09?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8Y2xpbmljfGVufDB8fDB8fA%3D%3D&w=1000&q=80"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://e-clinic.co.uk/wordpress_new/wp-content/uploads/2021/09/Image-14-09-2021-at-13.59.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://firstmedcenters.com/wp-content/uploads/2022/07/English-speaking-clinic-FirstMed-V2-2.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://www.extraspace.com/blog/wp-content/uploads/2020/10/preparing-home-for-airbnb.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://img1.wsimg.com/isteam/stock/4810/:/cr=t:0%25,l:5.83%25,w:88.34%25,h:100%25"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://healthwaymedical.com/wp-content/uploads/2022/01/Medico-Clinic-Surgery-1024x681.jpg"
              />
            </div>
            <div
              class="carousel-item carousel-item"
            >
              <img
                alt="travel vacation"
                class="d-block w-100"
                src="https://www.carepointmedical.com.au/wp-content/uploads/2020/08/doctors_three.jpg"
              />
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <div
        class="login-card col-4 pb-5 bg-white border rounded container"
      >
        <div
          class="logo-wrapper"
        >
          <img
            alt="logo"
            class="logo-img"
            height="200"
            src="LOGO-DOCTOR.png"
            width="200"
          />
        </div>
        <form
          class=""
        >
          <div
            class="mt-3 mb-3"
          >
            <input
              class="form-control"
              id="formBasicName"
              placeholder="Full name"
              required=""
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              id="formBasicEmail"
              placeholder="Email address"
              required=""
              type="email"
            />
            <small
              class="text-muted form-text"
            >
              We will never share your email with anyone else.
            </small>
          </div>
          <div
            class="mb-3 text-center"
          >
            <button
              class="btn btn-success"
              type="submit"
            >
              Register
            </button>
          </div>
        </form>
        <div>
          <p>
            Already have an account?
             
            <span>
              <button
                class="access-btn text-primary"
                type="button"
              >
                Login
              </button>
            </span>
          </p>
        </div>
      </div>
      <div
        class="carousel carousel slide"
      >
        <div
          class="carousel-indicators"
        >
          <button
            aria-current="true"
            aria-label="Slide 1"
            class="active"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 2"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 3"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 4"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 5"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 6"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 7"
            data-bs-target=""
            type="button"
          />
        </div>
        <div
          class="carousel-inner"
        >
          <div
            class="carousel-item active carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://images.unsplash.com/photo-1629909613654-28e377c37b09?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8Y2xpbmljfGVufDB8fDB8fA%3D%3D&w=1000&q=80"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://e-clinic.co.uk/wordpress_new/wp-content/uploads/2021/09/Image-14-09-2021-at-13.59.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://firstmedcenters.com/wp-content/uploads/2022/07/English-speaking-clinic-FirstMed-V2-2.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://www.extraspace.com/blog/wp-content/uploads/2020/10/preparing-home-for-airbnb.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://img1.wsimg.com/isteam/stock/4810/:/cr=t:0%25,l:5.83%25,w:88.34%25,h:100%25"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://healthwaymedical.com/wp-content/uploads/2022/01/Medico-Clinic-Surgery-1024x681.jpg"
            />
          </div>
          <div
            class="carousel-item carousel-item"
          >
            <img
              alt="travel vacation"
              class="d-block w-100"
              src="https://www.carepointmedical.com.au/wp-content/uploads/2020/08/doctors_three.jpg"
            />
          </div>
        </div>
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
